src/compiler/program.ts

```diff
@@ -411,8 +405,8 @@ namespace ts {
             }
 
             if (diagnostic) {
-                if (refFile) {
-                    diagnostics.add(createFileDiagnostic(refFile, start, length, diagnostic, ...diagnosticArgument));
+                if (refFile !== undefined && refEnd !== undefined && refPos !== undefined) {

```

Comment:
I think this may have been the cause of some issues I was having with "fancy" diagnostic reporting